src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/SNSConfiguration.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sns.SnsAsyncClient
+import java.net.URI
+
+@Configuration
+class SNSConfiguration(
+  @Value("\${aws.sns.provider}") private val provider: String,
+  @Value("\${aws.sns.region}") private val region: String,
+  @Value("\${aws.sns.access-key-id}") private val accessKeyId: String,
+  @Value("\${aws.sns.secret-access-key}") private val secretAccessKey: String,
+  @Value("\${aws.sns.endpoint.uri:}") private val uri: String,
+
+) {
+  @Bean
+  fun snsClient(): SnsAsyncClient {
```

Comment:
Would it help with tests down the line if we use the non-async [SnsClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sns/SnsClient.html)?

It seems to respond with a `PublishResponse` directly, which sounds OK since we're already firing this in an async listener.

Reply:
good idea.